groovy-language-server/src/main/java/com/palantir/groovylanguageserver/CompilerWrapper.java

```diff
@@ -37,17 +38,29 @@
 
     /**
      * Returns a mapping from absolute path of source file to symbols located within these source files.
-     * @return the list of symbols
      */
     Map<String, Set<SymbolInformation>> getFileSymbols();
 
+
+    /**
+     * Returns a mapping from a type name (class, interface, or enum) to symbols which reference those types.
+     */
+    Map<String, Set<SymbolInformation>> getTypeReferences();
+
+    /**
+     * Returns the locations of the symbols that reference the symbol defined by the given params.
+     * @param params the parameters used to filter down which symbol is referenced
+     * @return the set of locations
+     */
+    Set<SymbolInformation> findReferences(ReferenceParams params);
```

Comment:
The protocol wants an `Location[]` but it doesn't mention if it expects any sort of ordering. It's possible typefox returns a list for flexibility.